renderdoc/driver/d3d11/d3d11_context1_wrap.cpp

```diff
@@ -57,7 +57,8 @@ bool WrappedID3D11DeviceContext::Serialise_UpdateSubresource1(
   if(record && record->NumSubResources > (int)DstSubresource)
     record = (D3D11ResourceRecord *)record->SubResources[DstSubresource];
 
-  SERIALISE_ELEMENT_LOCAL(IsUpdate, bool(record->DataInSerialiser || IsActiveCapturing(m_State)))
+  SERIALISE_ELEMENT_LOCAL(IsUpdate,
+                          bool((record && record->DataInSerialiser) || IsActiveCapturing(m_State)))
```

Comment:
That change is fine, but to clarify record will be present while writing (the records are only used for tracking resources at capture time), so the check above on non-NULL is mostly there to protect the reading path. It would be better now to just move it inside the `if(ser.IsWriting())` path and remove the check I think.

It's harmless either way because the macro is kind of like a write-time initialisation so the second parameter will only be used during writing - otherwise not. I'm not sure whether the static analysis will pick that up though so there's no problem in checking it here.

Reply:
Ok. I think I got that.

Reply:
Blarg. Apparently I didn't get that right. I'm just going to drop that commit.

Reply:
Yeh this isn't quite what I meant - this bit can be moved inside the `if(ser.IsWriting())`

```
    if(record->NumSubResources > (int)DstSubresource)
      record = (D3D11ResourceRecord *)record->SubResources[DstSubresource];
```

But if you move the `SERIALISE_ELEMENT_LOCAL` inside then it will no longer be serialised when reading, causing mismatched serialisation. I also don't think this will compile anymore, since `IsUpdate` won't be in scope for the `if()` below.

Reply:
It will compile just fine on Linux. :)